all/test/all_test.go

```diff
@@ -552,6 +552,38 @@ func (s *TestSuite) TestAllCases() {
 			},
 			extraArgs: []string{"--with-validator"},
 		},
+		"typescript": {
+			lang: "typescript",
+			protofileName: "all/test/test.proto",
+			expectedOutputDir: "gen/pb-typescript",
+			fileExpectations: []FileExpectation{
+				{fileName: "Messages/MessageGrpc.ts"},
+				{fileName: "Messages/test_grpc.ts"},
+				{fileName: "Messages/test.ts"},
+			},
+		},
+		"typescript with alternative output dir": {
+			lang: "typescript",
+			protofileName: "all/test/test.proto",
+			expectedOutputDir: "gen/foo/bar",
+			fileExpectations: []FileExpectation{
+				{fileName: "Messages/MessageGrpc.ts"},
+				{fileName: "Messages/test_grpc.ts"},
+				{fileName: "Messages/test.ts"},
+			},
+			extraArgs: []string{"-o", "gen/foo/bar"},
+		},
+		"typescript with arguments": {
+			lang: "typescript",
+			protofileName: "all/test/test.proto",
+			expectedOutputDir: "gen/pb-typescript",
+			fileExpectations: []FileExpectation{
+				{fileName: "Messages/MessageGrpc.ts"},
+				{fileName: "Messages/test_grpc.ts"},
+				{fileName: "Messages/test.ts"},
+			},
+			extraArgs: []string{"--ts_opt", "useOptionals=messages"},
```

Comment:
Not particularly -- it seems out of scope to test that ts_opt on the ts-proto util works properly. I just wanted a test case to ensure that the passing of extra args itself doesn't break the addition, especially due to the special logical branch here: https://github.com/namely/docker-protoc/pull/317/files#diff-8eca9bfff91f2252cc45a47a4ec7d1687f750f70ebea1e5180d9912b2b63b0fdR389-R391

Reply:
right, I agree there is no point in testing `ts_opt` itself, but I figure we still want to test that the logic you linked is actually working, i.e that `--ts_opt` passes the value correctly to `ts_opt`.

Reply:
Yes, that makes sense, that would certainly add a level of security to the test suite.

 I think in order to do that I will have to find another command to pass that will generate extra files. `useOptionals=messages` just changes the way optional values are represented in the generated code (from `field: type | undefined` to `field?: type`), which I don't think would be possible (or at least not easy) to test against in this test suite.

I will take a look in a few days (unfortunately have just stepped away from development environment for a bit).

Reply:
sure, that makes sense. If however you do still want to use `useOptionals=messages`, you can find examples of reading one of the generated files and searching for a specific, unique string that would otherwise not appear.

Reply:
I see, I missed the tests that check actual file contents my first time looking. I've updated the test to include a check to ensure the flag is actually being passed and utilized

Reply:
Thanks!